include/core/shape.h

```diff
@@ -1300,11 +1302,19 @@ class RBFShape : public Shape<dim>
   std::shared_ptr<Rectangle<dim>> bounding_box;
   std::vector<size_t>             iterable_nodes;
 
-  std::map<const typename DoFHandler<dim>::cell_iterator, std::vector<size_t>>
+  std::map<const typename DoFHandler<dim>::cell_iterator,
+           std::shared_ptr<std::vector<size_t>>>
```

Comment:
Good question! They are shared_ptr because different cells may refer to the same vector of nodes. This allows the solver to save on the required memory, which is especially useful for heavy RBFs. The levels_not_precalculated parameter is the one to use for that